docs: mention container probes (#29085)
kubernetes/website: content/en/docs/concepts/workloads/pods/_index.md

This means that the Pods running on a node are visible on the API server,
but cannot be controlled from there.

## Container probes

> A Probe is a diagnostic performed periodically by the kubelet on a Container. To perform a diagnostic, the kubelet calls a Handler implemented by the container. There are three types of handlers:

Reviewer: Rule of thumb: if it's in the API but it's not listed by `kubectl api-resources`, either write it in backticks or in sentence case (not UpperCamelCase). Also, use italics when introducing new jargon.
So:
```suggestion
A _probe_ is a diagnostic performed periodically by the kubelet on a container. To perform a diagnostic, the kubelet can invoke different actions:
```

Author: Nice !! Learned new thing :)

> - ExecAction
> - TCPSocketAction
> - HTTPGetAction

Reviewer: How about:
```suggestion
- `ExecAction` (performed with the help of the container runtime)
- `TCPSocketAction` (checked directly by the kubelet)
- `HTTPGetAction` (checked directly by the kubelet)
```
?

Author: Looks perfect. I have added the suggested changes.

> learn more about it [here](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#container-probes)

Reviewer: ```suggestion
You can read more about [probes](/docs/concepts/workloads/pods/pod-lifecycle/#container-probes)
in the Pod Lifecycle documentation.
```

> You can read more about [probes](docs/concepts/workloads/pods/pod-lifecycle/#container-probes) in the Pod Lifecycle documentation.

Reviewer 2: ```suggestion
You can read more about [probes](/docs/concepts/workloads/pods/pod-lifecycle/#container-probes)
in the Pod lifecycle documentation.
```
